octodns/provider/dnsmadeeasy.py

```diff
@@ -265,9 +265,18 @@ def populate(self, zone, target=False, lenient=False):
         values = defaultdict(lambda: defaultdict(list))
         for record in self.zone_records(zone):
             _type = record['type']
-            if _type not in self.SUPPORTS:
-                self.log.warning('populate: skipping unsupported %s record',
-                                 _type)
+            data_for = getattr(self, '_data_for_{}'.format(_type), None)
+            if not data_for or not self.supports(
+                Record.new(
+                    zone,
+                    record['name'],
+                    data_for(_type, [record, ])
+                )
```

Comment:
Creating the full `Record` here will mean that any validation problems with the data that comes down will cause exceptions and won't be ignored as they normally would be lenient. From what I can tell by looking at `supports()` there doesn't seem to be anything that strictly requires it to be a `Record` object. Can that method be refactor to just pass in the necessary data, probably the output of `data_for`, and work with that?

This also results in double-running `data_for` and create the `Record` object twice.  It may make sense to just leave this block as it originally was and then go down into the for that's creating records below and add in a call to `supports()` around here https://github.com/octodns/octodns/pull/668/files#diff-f9649e542e179d4617a1512b9cedc54e1fd1b0d4e680a19150c8f2458e1208c9R289. It should be fine to "skip" records down there too.

Reply:
changes() in zone.py will emit the warning about the unsupported record at a later stage, so that code has now been refactored away.